v0.5/fastn-p1/src/parser/section_init.rs

```diff
@@ -22,6 +22,7 @@ mod test {
         t!("-- foo: ", {"name": {"name": "foo"}}, " ");
         t!("-- foo: hello", {"name": {"name": "foo"}}, " hello");
         t!("-- integer foo: hello", {"name": {"name": "foo", "kind": "integer"}}, " hello");
+        ti!("-- integer héllo: foo", {"name": {"name": "héllo", "kind": "integer"}}, " foo", 17, 18);
```

Comment:
Why do we have to check 17/18 here? The `assert` on `remaining` should be enough.

Reply:
Basically `remaining` ensures if the value of `Scanner.index` is correct, but not `Scanner.s_index`.

Reply:
Pass original source to `Scanner.remaining()`, and make it assert that remaining string generated by current char approach, and sub slice of original passed source are the same, and then it remains either.

Reply:
`.remaining()` is `#[cfg(test)]` so this extra assertion / work is fine.